### PR TITLE
Session03 homework

### PR DESCRIPTION
**Pull request by dcohashi**
Here is the main homework and the directory as html links.

### CODE REVIEW

File: resources/session03/wsgi/calculator.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
+
+
+def num(s):
+    # return some sort of number from a string
+    try:
+        return int(s)
+    except ValueError:
+        return float(s)
```
Comment (cewing) [inline resources/session03/wsgi/calculator.py]: What happens with this if I give it the string `a` as input?

File: resources/session03/wsgi/calculator.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+def add(arg1, arg2):
+    return("{} + {} = {}".format(arg1, arg2, arg1+arg2))
+
+
+def subtract(arg1, arg2):
+    return("{} - {} = {}".format(arg1, arg2, arg1-arg2))
+
+
+def divide(arg1, arg2):
+    return("{} / {} = {}".format(arg1, arg2, arg1/arg2))
```
Comment (cewing) [inline resources/session03/wsgi/calculator.py]: These four functions feel a bit repetitive to me.  The only real difference between them is the name of the operation being performed (`+`, `-`, `*`, or `/`) and the operator used.  Could you think of a way to reduce the four functions to just one?  Look at the `operator` dict you declare below this and think about what you might do instead.  If you knew about the [operator module](https://docs.python.org/3.5/library/operator.html) from the Python standard library, would it make a difference in your approach?

File: resources/session03/wsgi/calculator.py [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    '''
+    matchpath = path.lstrip('/')
+    parts = matchpath.split('/')
+    try:
+        func = operator[parts[0]]
+    except KeyError:
+        raise NameError
+    arg1 = num(parts[1])
+    arg2 = num(parts[2])
+    return(func, arg1, arg2)
```
Comment (cewing) [inline resources/session03/wsgi/calculator.py]: Nice adaptation of the dispatch function.  Well done.

File: resources/session03/wsgi/calculator.py [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+    Return function based on the operator and return operands in a list
+    '''
+    matchpath = path.lstrip('/')
+    parts = matchpath.split('/')
+    try:
+        func = operator[parts[0]]
+    except KeyError:
+        raise NameError
+    arg1 = num(parts[1])
+    arg2 = num(parts[2])
```
Comment (cewing) [inline resources/session03/wsgi/calculator.py]: Your validation function above is still a bit fragile, think about what that means for this code here.

File: resources/session03/wsgi/calculator.py [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+    try:
+        path = environ.get('PATH_INFO', None)
+        if path is None:
+            raise NameError
+        func, arg1, arg2 = resolve_path(path)
+        status = "200 OK"
+        body = func(arg1, arg2)
+    except (ZeroDivisionError, ValueError):
+        status = "400 Bad Request"
+        body = "<h1>Bad Request<h1>"
```
Comment (cewing) [inline resources/session03/wsgi/calculator.py]: good work here.

**Comment by cewing** [comment]
Nice job on this one.  There are improvements that could be made, but the app works as is, for some limited set of URLs.  It's a bit unfortunate that I have to manually enter a URL to get a calculation to happen,  how might you solve that problem?

File: resources/session03/wsgi/calculator.py (hunk 1 above)
Comment (dcohashi) [inline resources/session03/wsgi/calculator.py]: I actually tried this function with an invalid string.  It returned a ValueError that I tried to catch in the calling function.  Should I be catching it in this local function instead?

Sent from my iPad

On Jan 25, 2016, at 9:31 PM, Cris Ewing <notifications@github.com> wrote:

In resources/session03/wsgi/calculator.py:

> @@ -0,0 +1,79 @@
> +
> +
> +def num(s):
> +    # return some sort of number from a string
> +    try:
> +        return int(s)
> +    except ValueError:
> +        return float(s)
What happens with this if I give it the string a as input?

**Comment by dcohashi** [comment]
Thank you for the feedback.  It is really helpful

Sent from my iPad

On Jan 25, 2016, at 9:36 PM, Cris Ewing <notifications@github.com> wrote:

Nice job on this one. There are improvements that could be made, but the app works as is, for some limited set of URLs. It's a bit unfortunate that I have to manually enter a URL to get a calculation to happen, how might you solve that problem?

**Pull request closed without merging**